msal/src/main/java/com/microsoft/identity/nativeauth/INativeAuthPublicClientApplication.kt

```diff
@@ -71,89 +69,47 @@ interface INativeAuthPublicClientApplication : IPublicClientApplication {
      * @return [com.microsoft.identity.nativeauth.statemachine.results.SignInResult] see detailed possible return state under the object.
      * @throws [MsalException] if an account is already signed in.
      */
-    fun signIn(username: String, scopes: List<String>? = null, callback: NativeAuthPublicClientApplication.SignInCallback)
-
-    /**
-     * Sign in the account using username and password; Kotlin coroutines variant.
-     *
-     * @param username username of the account to sign in.
-     * @param password password of the account to sign in.
-     * @param scopes (Optional) list of scopes to request.
-     * @return [com.microsoft.identity.nativeauth.statemachine.results.SignInUsingPasswordResult] see detailed possible return state under the object.
-     * @throws MsalClientException if an account is already signed in.
-     */
-    suspend fun signInUsingPassword(username: String, password: CharArray, scopes: List<String>? = null): SignInUsingPasswordResult
-
-    /**
-     * Sign in the account using username and password; callback variant.
-     *
-     * @param username username of the account to sign in.
-     * @param password password of the account to sign in.
-     * @param scopes (Optional) list of scopes to request.
-     * @param callback [com.microsoft.identity.nativeauth.NativeAuthPublicClientApplication.SignInUsingPasswordCallback] to receive the result.
-     * @return [com.microsoft.identity.nativeauth.statemachine.results.SignInUsingPasswordResult] see detailed possible return state under the object.
-     * @throws MsalClientException if an account is already signed in.
-     */
-    fun signInUsingPassword(username: String, password: CharArray, scopes: List<String>? = null, callback: NativeAuthPublicClientApplication.SignInUsingPasswordCallback)
+    fun signIn(username: String, password: CharArray? = null, scopes: List<String>? = null, callback: NativeAuthPublicClientApplication.SignInCallback)
 
     /**
      * Sign up the account starting from a username; Kotlin coroutines variant.
      *
      * @param username username of the account to sign up.
+     * @param password password of the account to sign up.
      * @param attributes (Optional) user attributes to be used during account creation.
      * @return [com.microsoft.identity.nativeauth.statemachine.results.SignUpResult] see detailed possible return state under the object.
      * @throws MsalClientException if an account is already signed in.
      */
-    suspend fun signUp(username: String, attributes: UserAttributes? = null): SignUpResult
+    suspend fun signUp(username: String, password: CharArray? = null, attributes: UserAttributes? = null): SignUpResult
 
     /**
      * Sign up the account starting from a username; callback variant.
      *
      * @param username username of the account to sign up.
+     * @param password password of the account to sign up.
```

Comment:
let's make it explicit that this is optional, like we do for attributes.

Reply:
done